src/components/allCommunities/CommunityCard.tsx

```diff
@@ -1,25 +1,63 @@
-import { Heading, Image, Stack, Tag } from "@chakra-ui/react"
+import { Heading, Image, Portal, Stack, Tag } from "@chakra-ui/react"
+import { useWeb3React } from "@web3-react/core"
 import Card from "components/common/Card"
 import { Link } from "components/common/Link"
-import { CommunityProvider } from "components/community/Context"
-import type { Community } from "temporaryData/communities"
+import { useCommunity } from "components/community/Context"
+import { useLevelAccess } from "components/community/Levels/components/Level/hooks/useLevelAccess"
+import { MutableRefObject, useMemo } from "react"
+import useSWR from "swr"
 
 type Props = {
-  community: Community
+  refYours: MutableRefObject<HTMLDivElement>
+  refAccess: MutableRefObject<HTMLDivElement>
+  refOther: MutableRefObject<HTMLDivElement>
 }
 
-const CommunityCard = ({ community }: Props): JSX.Element => {
-  const membersCount = community.levels
+const getJoinedCommunities = async (
+  _: string,
+  address: string
+): Promise<Array<number>> =>
+  fetch(`${process.env.NEXT_PUBLIC_API}/getUserMembership/${address}`).then(
+    (response) => {
+      if (response.ok) {
+        return response.json().then((data) => data.communities)
+      }
+      return []
+    }
+  )
+
+const CommunityCard = ({ refYours, refOther, refAccess }: Props): JSX.Element => {
+  const {
+    levels,
+    urlName,
+    imageUrl,
+    name: communityName,
+    chainData: {
+      token: { symbol: tokenSymbol },
+    },
+    id,
+  } = useCommunity()
+  const { account } = useWeb3React()
+  const [hasAccess] = useLevelAccess(levels[0].accessRequirement)
+  const { data: joinedCommunitites } = useSWR(
+    ["joined_communities", account],
+    getJoinedCommunities
+  )
+  const isMember = joinedCommunitites?.includes(id)
+
+  const containerRef = useMemo(() => {
+    if (isMember) return refYours
+    if (hasAccess) return refAccess
+    return refOther
+  }, [isMember, hasAccess])
```

Comment:
shouldn't it include the refs too?

Reply:
I don't think it's necessary, by the time the `useEffect` runs, the refs already have the Grid elements as values, and they don't change after that. We can include them just in case, I don't think it will affect anything